scripts/setup-seafile-mysql.py

```diff
@@ -1227,11 +1231,129 @@ def set_file_perm():
+
+    if db_config.use_existing_db:
+        db_config.check_mysql_user(mysql_user, mysql_user_passwd)
+        db_config.seafile_mysql_user = mysql_user
+        db_config.seafile_mysql_password = mysql_user_passwd
+        db_config.ccnet_db_name = db_config.validate_db_name(ccnet_db)
+        db_config.seafile_db_name = db_config.validate_db_name(seafile_db)
+        db_config.seahub_db_name = db_config.validate_db_name(seahub_db)
+    else:
+        if db_config.mysql_host != '127.0.0.1' and not mysql_user_host:
+            raise InvalidParams('In create db mode, miss mysql user host parameter')
+        if not mysql_user_host:
+            db_config.mysql_userhost = 'localhost'
+        else:
+            db_config.mysql_userhost = db_config.validate_mysql_user_host(mysql_user_host)
+
+        if not mysql_root_passwd:
+            raise InvalidParams('In create db mode, miss mysql root passwd parameter')
+        db_config.root_password = db_config.validate_root_passwd(mysql_root_passwd)
+
+        if mysql_user == 'root':
+            db_config.seafile_mysql_user = 'root'
+            db_config.seafile_mysql_password = db_config.root_password
+        else:
+            if db_config.mysql_user_exists(mysql_user):
+                db_config.check_mysql_user(mysql_user, mysql_user_passwd)
+            db_config.seafile_mysql_user = mysql_user
+            db_config.seafile_mysql_password = mysql_user_passwd
+        db_config.ccnet_db_name = ccnet_db
+        db_config.seafile_db_name = seafile_db
+        db_config.seahub_db_name = seahub_db
+
+    global need_pause
+    need_pause = False
+
 
 def main():
+    if len(sys.argv) > 2 and sys.argv[1] == 'auto':
+        sys.argv.remove('auto')
+        parser = argparse.ArgumentParser()
+        parser.add_argument('-n', '--server-name', help='server name')
+        parser.add_argument('-i', '--server-ip', help='server ip or domain')
+        parser.add_argument('-p', '--fileserver-port', help='fileserver port')
+        parser.add_argument('-d', '--seafile-dir', help='seafile dir to store seafile data')
+        parser.add_argument('-e', '--use-existing-db',
+                            help='use mysql existing db, 1: create new 2: use exist')
+        parser.add_argument('-o', '--mysql-host', help='mysql host')
+        parser.add_argument('-t', '--mysql-port', help='mysql port')
+        parser.add_argument('-u', '--mysql-user', help='mysql user')
+        parser.add_argument('-w', '--mysql-user-passwd', help='mysql user password')
+        parser.add_argument('-q', '--mysql-user-host', help='mysql user host')
+        parser.add_argument('-r', '--mysql-root-passwd', help='mysql root password')
+        parser.add_argument('-c', '--ccnet-db', help='ccnet db')
+        parser.add_argument('-s', '--seafile-db', help='seafile db')
+        parser.add_argument('-b', '--seahub-db', help='seahub db')
+
+        args = parser.parse_args()
+        check_params(args)
```

Comment:
Catch `InvalidParamsException` here, and only print the error message, rather than printing the full traceback like these:

 ```
Traceback (most recent call last):
  File "setup-seafile-mysql.py", line 1425, in <module>
    main()
  File "setup-seafile-mysql.py", line 1351, in main
    check_params(args)
  File "setup-seafile-mysql.py", line 1311, in check_params
    raise InvalidParams('In create db mode, miss mysql root passwd parameter')
__main__.InvalidParams: In create db mode, miss mysql root passwd parameter
```